src/Badges.sol

```diff
@@ -69,16 +85,86 @@ contract Badges is
   // Give is called by someone who has authority to create badge specs
   // Prior to calling "Give", the "to" address would have alrady requested
   // the badge (like joining a wait list)
+  function giveExpiringBadge(
+    address _to,
+    string calldata _uri,
+    bytes calldata _signature,
+    string calldata _expirationType,
+    uint256 _expirationValue
+  ) external virtual returns (uint256) {
+    require(msg.sender != _to, "giveExpiringBadge: cannot give from self");
+
+    uint256 voucherHashId = safeCheckExpiringBadge(
+      msg.sender,
+      _to,
+      _uri,
+      _signature,
+      _expirationType,
+      _expirationValue
+    );
+    uint256 tokenId = mint(_to, _uri, 0);
+    usedHashes.set(voucherHashId);
+    voucherHashIds[tokenId] = voucherHashId;
```

Comment:
Could we refactor some of this code?

Reply:
There's some code reused across both give/take functions for expiring badge. 
Or consider moving the setting of voucherHashId logic into mint(..) since it is used by all give/take functions